pending_review/pull_request.go

```diff
@@ -176,27 +191,58 @@ func (s *PullRequestService) determineTypeOfChange(ctx context.Context, owner st
 		return nil, resp, fmt.Errorf("%w", ErrInvalidChange)
 	}
 
-	change, err := getDiff(files[0])
+	change, err := processChangedFiles(files)
 	if err != nil {
 		return nil, resp, err
 	}
 
+	return change, resp, nil
+}
+
+func processChangedFiles(files []*CommitFile) (*change, error) {
+	if len(files) < 1 {
+		return nil, fmt.Errorf("%w", ErrInvalidChange)
+	}
+
+	change, err := getDiff(files[0])
+	if err != nil {
+		return nil, err
+	}
+
+	addition := files[0].GetAdditions()
+	deletions := files[0].GetDeletions()
 	for _, file := range files[1:] {
 		obtained, err := getDiff(file)
 		if err != nil {
-			return nil, resp, err
+			return nil, err
 		}
 
-		if change.Recipe != obtained.Recipe { // PR should only be changing one recipe at a time
-			return nil, resp, fmt.Errorf("%w", ErrInvalidChange)
+		if change.Recipe != obtained.Recipe {
+			return nil, fmt.Errorf("%w", ErrInvalidChange)
 		}
 
 		if change.Change == NEW && obtained.Change == EDIT {
-			change.Change = EDIT // Any edit breaks the "new recipe" definition
+			change.Change = EDIT
 		}
+
+		addition += file.GetAdditions()
+		deletions += file.GetDeletions()
 	}
 
-	return change, resp, nil
+	//if len(files) <= 2 && addition <= 10 && deletions == 0 {
+	if len(files) <= 2 && (addition+deletions) <= 10 {
+		change.Weight = TINY
+	} else if len(files) <= 4 && (addition+deletions) <= 25 {
+		change.Weight = SMALL
+	} else if len(files) <= 7 && (addition+deletions) <= 100 {
+		change.Weight = REGULAR
+	} else if len(files) <= 9 && addition <= 225 && deletions == 0 { // Basic new recipe addition with `test_v1_package`
+		change.Weight = REGULAR
+	} else if len(files) > 12 || (addition+deletions) >= 500 {
+		change.Weight = TOO_MUCH
+	}
```

Comment:
```suggestion
	} else if len(files) <= 12 || (addition+deletions) < 500 {
		change.Weight = HEAVY
	} else {
		change.Weight = TOO_MUCH	
	}
```

Reply:
this more of a readability change than anything else, because I believe change.weight is initialized to HEAVY by default.